lib/deploy/lifecycleHooks.js

```diff
@@ -1,43 +1,78 @@
 'use strict';
 
+var _ = require('lodash');
 var RSVP = require('rsvp');
 var utils = require('../utils');
 var chalk = require('chalk');
-var process = require('child_process');
+var cprocess = require('child_process');
 var FirebaseError = require('../error');
+var getProjectId = require('../getProjectId');
+var logger = require('../logger');
+
+
+function runCommand(command, options) {
+  return new Promise(function(resolve, reject) {
+    var child = cprocess.spawn(command, _.assign({}, options, {
+      shell: true,
+      stdio: [0, 1, 2] // Inherit STDIN, STDOUT, and STDERR
+    })
```

Comment:
nit: The extra line of `);` without even indentation isn't helping readability. I would prefer one of following code styles:

```js
// Combining all closing symbols because there's only one level of indent
var child = cprocess.spawn(command, _.assign({}, options, {
    shell: true,
    stdio: [0, 1, 2] // Inherit STDIN, STDOUT, and STDERR
}));
```

```js
// Adding a second layer of indentation for clarity
var child = cprocess.spawn(
  command,
  _.assign({}, options,  {
      shell: true,
      stdio: [0, 1, 2] // Inherit STDIN, STDOUT, and STDERR
  }));
```

```js
// Avoiding style guide edge cases:
var spawnOpts = _.assign({}, options, {
  shell: true,
  stdio: [0, 1, 2] // Inherit STDIN, STDOUT, and STDERR
});
var child = cprocess.spawn(command, spawnOpts);
```